suzieq/poller/controller/source/native.py

```diff
@@ -28,11 +28,14 @@ def _validate_config(self, input_data: dict):
         self._valid_fields.extend(['hosts'])
         super()._validate_config(input_data)
 
-        if not isinstance(input_data.get('hosts'), list):
-            raise InventorySourceError(f'{self._name} Hosts must be a list')
-
         if not input_data.get('hosts'):
-            raise InventorySourceError(f'{self._name} Hosts must not be empty')
+            raise InventorySourceError(f"{self._name} The 'hosts' field with "
+                                       "the list of nodes to poll is mandatory"
+                                       )
+
+        if not isinstance(input_data.get('hosts'), list):
```

Comment:
Same error as what @claudiolor pointed out in the other case? You have to check its not an empty list?

Reply:
The check is done in the line above. I'm doing `if not input_data.get('hosts'): #error`. This check wiil catch both invalid and empty list.